aten/src/ATen/native/cpu/SoftMaxKernel.cpp

```diff
@@ -207,6 +207,146 @@ struct vec_host_softmax_lastdim {
   }
 };
 
+#if defined(CPU_CAPABILITY_AVX2) && !defined(_MSC_VER)
+inline void _vec_softmax(
```

Comment:
What prevents this from becoming an instantiation of the other existing implementations such as _vec_softmax_lastdim?

Reply:
It shouldn't have a problem. Remove the judgement of these macros.

Reply:
Is it feasible to merge this with the other implementation to reduce code size in this PR? Or should we do this as a follow up?

Reply:
Hi @cpuhrsch, I think we should keep this BF16 specialized template for better performance and accuracy. https://github.com/pytorch/pytorch/blob/2aebb58345f09b23160506611ba28fcfd053f2f4/aten/src/ATen/native/cpu/SoftMaxKernel.cpp#L210-L215 The bf16 datatype can share the implementation with general datatype(fp32, double). But the performance may not be good since too many implicit cast exists. https://github.com/pytorch/pytorch/blob/2aebb58345f09b23160506611ba28fcfd053f2f4/aten/src/ATen/native/cpu/SoftMaxKernel.cpp#L346-L352